mirrord/sip/src/lib.rs

```diff
@@ -1,3 +1,31 @@
+//! This crate concerns MacOS' SIP (System Integrity Protection) mechanism and how to sidestep it.
+//!
+//! ### Why?
+//!
+//! SIP makes it such that we are unable to load the `mirrord-layer` lib dynamically into the user's
+//! process onto MacOS machines. This is ostensibly a security measure to prevent the malicious or
+//! unintended modification of processes, but mirrord's dynamic library loading is neither. To get
+//! around this, we patch the user's process by creating a new unprotected version and signing it
+//! ourselves. Then we can load the layer dynamically into our unprotected copy and continue as
+//! usual. If the process is a script, the shebang must be pointed to the patched version of the
```

Comment:
Special cases (like the script process in this case) should probably get a `-` or be a bit separated from the general explanation, like those info-panels you see in books (maybe just a `-` list here, but I can see it being a whole sub-header in some cases, so let's not be too rigid).